### PR TITLE
Scripts to check issue 2482

### PR DESCRIPTION
**Pull request by HSavynetska**
ATF Test Scripts to check [#2482](https://github.com/smartdevicelink/sdl_core/issues/2482)

This PR is **not ready** for review.
This is **draft** version of the script.

### Summary
A test script to verify that issue is fixed

### ATF version
[develop](https://github.com/smartdevicelink/sdl_atf/tree/develop)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  helpPrompt = {
+    { type = "TEXT", text = "pathToFile2" }
+  },
+  timeoutPrompt = {
+    { type = "TEXT", text = "pathToFile3" }
+  },
+  interactionLayout = "KEYBOARD"
+}
+
+local performHmiParams = {
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua]: There is not the need to define performHmiParams, values for check in sendPerformInteraction_SUCCESS  can be used from performParams.

File: test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("Register App", common.registerApp)
+runner.Step("Activate App", common.activateApp)
+
+runner.Title("Test")
+for _, v in ipairs(keyPressMode) do
+	runner.Step("SetGlobalProperties with keypressMode " .. v, setGlobalProperties, { v })
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua]:
```
runner.Step("CreateInteractionChoiceSet", createInteractionChoiceSet)
runner.Step("Send PerformInteraction SUCCESS response", sendPerformInteraction_SUCCESS)
```
these functions also must be executed in the loop for different keypress modes, in current implementation there is no reason of the loop and PI will be performed with RESEND_CURRENT_ENTRY only

File: test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+-- 	  edit Keyboard properties "KeypressMode as Single_keyPress" and press ok
+-- 5) Select 'Send Message' in the test app on the phone and
+--    Select CreateInteractionChoiceSet then enter all the details and press OK.
+-- 6) Select 'Send Message' in the test app on the phone and
+--    Select "PerformInteraction", Enter data for Initial TextInitial Prompt:
+--    Interaction Mode: MANUAL_ONLY, Layout: KEYBOARD(as in step 4 KeyPress mode: SINGLE_KEYPRESS )and press OK
+-- 7) Verify the Screen displayed on SYNC
+-- 8) Press on the text box and enter the Single key in the keyboard displayed on the sync
+-- Expected result:
+-- 1) SUCCESS response should be recieved for both SetGlobalProperties and PerformInteraction RPC.
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua]: typo recieved

File: test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+-- Actual result:
+-- UNSUPPORTED_RESOURCE response is recieved fot SetGlobalProperties.
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua]: typo recieved fot

File: test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("Register App", common.registerApp)
+runner.Step("Activate App", common.activateApp)
+runner.Step("CreateInteractionChoiceSet", createInteractionChoiceSet)
+
+for _, v in pairs(keyPressMode) do
+  runner.Title("Test")
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua]: move title before loop

File: test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+	runner.Step("SetGlobalProperties with keypressMode " .. v, setGlobalProperties, { v })
+  runner.Step("Send PerformInteraction SUCCESS response", sendPerformInteraction_SUCCESS)
+
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua]: remove redundant new line

File: test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+--[[ Local Functions ]]
+local function setGlobalProperties(pkeyPressMode)
+  local cid = common.getMobileSession():SendRPC("SetGlobalProperties",
+  {
+    keyboardProperties = {
+      keypressMode = pkeyPressMode
+    }
+  })
+  common.getHMIConnection():ExpectRequest("UI.SetGlobalProperties",
+    { appID = common.getHMIAppId(),
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua]: please update alignment in this expectation

File: test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("Register App", common.registerApp)
+runner.Step("Activate App", common.activateApp)
+runner.Step("CreateInteractionChoiceSet", createInteractionChoiceSet)
+
+for _, v in pairs(keyPressMode) do
+  runner.Title("Test")
+	runner.Step("SetGlobalProperties with keypressMode " .. v, setGlobalProperties, { v })
+  runner.Step("Send PerformInteraction SUCCESS response", sendPerformInteraction_SUCCESS)
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2482_UNSUPPORTED_RESOURCE_response_is_received_for_SetGlobalProperties_RPC_when_KEYPRESS_MODE_is_set_to_SINGLE_KEYPRESS_or_QUEUE_KEYPRESSES_or_RESEND_CURRENT_ENTRY.lua]: Please  concatenate keypressMode to test case name.

**Comment by dboltovskyi** [comment]
Livio team please do not review this PR for now since it hasn't passed our internal review yet.  
Review and appropriate updates going be done till 1st of November.
Thank you.

**Comment by dboltovskyi** [comment]
The internal review iteration has been moved to the LuxoftSDL.

**Pull request closed without merging**